### PR TITLE
Ensure all folders are accessible by appuser in the Docker container

### PR DESCRIPTION
**Pull request by agpituk**
# What's changing
I run into an error trying to run the container, `/home/appuser/.cache` is not accessible for appuser

# How to test it
Build the container and try to run it locally
# I already...

- [X] Tested the changes in a working environment to ensure they work as expected
- [N/A ] Added some tests for any new functionality
- [N/A ] Updated the documentation (both comments in code and under `/docs`)

### CODE REVIEW

**Review by daavoo — COMMENTED** [review]
That is strange I didn't run into the error when running inside codespaces.

High level question, do we actually need `appuser`? I just added it from this example https://docs.streamlit.io/deploy/tutorials/kubernetes#create-a-dockerfile

**Review by daavoo — APPROVED** [review]

**Review by daavoo — COMMENTED** [review]
build is failing:

https://github.com/mozilla-ai/document-to-podcast/actions/runs/12314852998/job/34371817672

File: Dockerfile [hunk 1]
```diff
@@ -16,7 +16,9 @@ RUN pip3 install /home/appuser/document-to-podcast
 RUN python3 demo/download_models.py
```
Comment (daavoo) [inline Dockerfile:16]: Ok I think the problem might be here. 
Inside this script, HuggingFace is downloading the models to `~/.cache` here but at this point `~` doesn't point to `/home/appuser`. 

I think if we set `USER appuser` before running `demo/download_models.py` the error should be fixed

Comment (agpituk) [reply Dockerfile:16]: that makes sense :) I'll do that and test! thanks David!

**Comment by daavoo** [comment]
Ok I made some changes so that models are correctly downloaded after configuring the user

**Pull request merged**